frappe/database/query.py

```diff
@@ -587,35 +628,39 @@ def join_(self, criterion, fields, table, join):
 					and str(field).startswith("`tab")
 					and (f"`tab{table}`" not in str(field))
 				):
-					join_table = table_from_string(str(field))
-					if self.fieldname:
-						criterion = criterion.left_join(join_table).on(
-							getattr(join_table, "name") == getattr(frappe.qb.DocType(table), self.fieldname)
-						)
-					else:
-						criterion = criterion.left_join(join_table).on(
-							getattr(join_table, "parent") == getattr(frappe.qb.DocType(table), "name")
-						)
 					has_join = True
-
-			if has_join:
-				for idx, field in enumerate(fields):
-					if not is_pypika_function_object(field):
-						field = field if isinstance(field, str) else field.get_sql()
-						if not TABLE_PATTERN.search(str(field)):
-							fields[idx] = getattr(frappe.qb.DocType(table), field)
-					else:
-						field.args = [getattr(frappe.qb.DocType(table), arg.get_sql()) for arg in field.args]
-						field.args[0] = getattr(frappe.qb.DocType(table), field.args[0].get_sql())
-						fields[idx] = field
+					join_table = table_from_string(str(field))
```

Comment:
maybe try to break this down into simpler functions as well if possible :P

Reply:
done